frontend/src/bundles/common/components/sort-button/sort-button.tsx

```diff
@@ -0,0 +1,66 @@
+import clsx from 'clsx';
+import {
+    type ButtonHTMLAttributes,
+    type ReactNode,
+    useCallback,
+    useEffect,
+    useState,
+} from 'react';
+
+import { IconName } from '../../enums/enums';
+import { Icon } from '../components';
+import styles from './styles.module.scss';
+
+type SortType = 'asc' | 'desc' | null;
+
+interface Properties
+    extends Omit<ButtonHTMLAttributes<HTMLButtonElement>, 'onClick'> {
+    initialSortType?: SortType;
+    children: ReactNode;
+    className?: string;
+    onSort: (sortMethod: SortType) => void;
+}
+
+const SortButton = ({
+    children,
+    initialSortType = null,
+    className = '',
+    onSort,
+}: Properties): JSX.Element => {
+    const [sortMethod, setSortMethod] = useState<SortType>(initialSortType);
+
+    const handleSortClick = useCallback(() => {
+        if (sortMethod === 'desc') {
+            setSortMethod(null);
+        } else {
+            setSortMethod((sortMethod === 'asc' ? 'desc' : null) ?? 'asc');
```

Comment:
Avoid using nested conditions

Reply:
But without this ternary operator, the only way to implement the required functionality is to create multiple if statements. I don't think this is a good solution. So how about leaving this operator and moving on?

Reply:
I believe that you can find a solution